dask/dataframe/io/parquet.py

```diff
@@ -540,16 +582,174 @@ def _read_pyarrow(fs, fs_token, paths, columns=None, filters=None,
                              out_type == Series,
                              dataset.partitions,
                              categories)
-            for i, piece in enumerate(dataset.pieces)
+            for i, piece in enumerate(non_empty_pieces)
         }
     else:
         meta = strip_unknown_categories(meta)
-        divisions = (None, None)
         task_plan = {(task_name, 0): meta}
 
     return out_type(task_plan, task_name, meta, divisions)
 
 
+def _nat_sorted_pieces(pieces):
+    """
+    Return list of pyarrow Parquet pieces sorted 'naturally' by path
+
+    Lexicographic sorted list of paths:
+    ['/.../part.1.parquet', '/.../part.10.parquet', '/.../part.2.parquet']
+
+    Naturally sorted list of paths:
+    ['/.../part.1.parquet', '/.../part.2.parquet', '/.../part.10.parquet']
+
+    Parameters
+    ----------
+    pieces : list[pyarrow.parquet.ParquetDatasetPiece]
+
+    Returns
+    -------
+    list[pyarrow.parquet.ParquetDatasetPiece]
+    """
+    def to_sort_tuple(piece):
+        return [int(s) if s.isdigit() else s for s in re.split('(\d+)', piece.path)]
+
+    return sorted(pieces, key=to_sort_tuple)
```

Comment:
This will maybe be useful elsewhere?  I suggest that we make this non-parquet specific and throw it in `dask/utils.py`.  Perhaps it should consume a sequence of filenames?

Reply:
I moved the core of this to `dask.utils` as `nat_sort_key`